tools/tpm2_listpersistent.c

```diff
@@ -89,20 +95,20 @@ int tpm2_tool_onrun(TSS2_SYS_CONTEXT *sapi_context, tpm2_option_flags flags) {
                                    &capabilityData, 0 );
     if(rval != TPM_RC_SUCCESS)
     {
-        printf("\n......GetCapability: Get persistent object list Error."
-               " TPM Error:0x%x......\n", rval);
+        LOG_ERR("\n......GetCapability: Get persistent object list Error."
+               " TPM Error:0x%x......", rval);
         return 1;
     }
 
-    printf( "%d persistent objects defined.\n", capabilityData.data.handles.count);
+    tpm2_tool_output( "%d persistent objects defined.", capabilityData.data.handles.count);
     UINT32 i;
     for( i=0; i < capabilityData.data.handles.count; i++ )
     {
-        printf("\n  %d. Persistent handle: 0x%x\n", i, capabilityData.data.handles.handle[i]);
+        tpm2_tool_output("\n  %d. Persistent handle: 0x%x\n", i, capabilityData.data.handles.handle[i]);
```

Comment:
no caps, just "persistent handle" maybe do array notation:

persistent handle[0]:0x12345678

Would spaces in the output add trailing/leading whitespace to variables in bash?

Reply:
We'd probably want readPublic to have easier syntax to parse...
```
persistent-handle[0]:0x12345678 key-alg:sha1 hash-alg:sha256 attrs:0x56
``
This way to get key-alg for a handle...
```
command | cut -d' ' -f 2-2 | cut -d\: -f 2-2
sha1
```

Reply:
@williamcroberts the intention of this patch was just to replace the printf() calls for the tool output and log err, but I do agree with you that the current output is far from ideal: not easier for machines to parse yet no really readable for humans either.

So I'll add another commit on top that takes your comments into account and changes the tool output. I likely do it tomorrow though since now I'm busy with other stuff.

Reply:
@martinezjavier that sounds perfect.